packages/fxa-settings/src/index.tsx

```diff
@@ -30,25 +23,43 @@ try {
   if (!config.servers.gql.url) {
     throw new Error('GraphQL API URL not set');
   }
+  if (!config.servers.auth.url) {
+    throw new Error('Auth Server URL not set');
+  }
+  const authClient = new AuthClient(config.servers.auth.url)
+
+  // sessionToken can change due to a password change so we
+  // can't simply add it in the constructor
+  const httpLink = createHttpLink({
+    uri: `${config.servers.gql.url}/graphql`,
+  });
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        Authorization: getSessionToken(),
+      }
+    }
+  });
 
   // TODO: due to the nature and specificity of the requests we need to talk
   // directly to the auth server, not the GQL server, for the following routes:
   // - POST /password/change/start
   // - POST /password/change/finish
   // - POST /account/destroy
   const apolloClient = new ApolloClient({
-    cache: new InMemoryCache(),
-    uri: `${config.servers.gql.url}/graphql`,
-    headers: {
-      authorization: getSessionToken(),
-    },
+    cache,
+    link: authLink.concat(httpLink),
+    typeDefs
   });
   const queryParams = searchParams(window.location.search);
 
   render(
     <React.StrictMode>
       <ApolloProvider client={apolloClient}>
-        <App {...{ queryParams }} />
+        <AuthContext.Provider value={{client: authClient}}>
```

Comment:
here we set the authClient into a react context. idk if context is the best pattern for it yet but i started by mimicking the apollo client